### ISSUE TITLE
Large ellipses tend to under/overflow bounding box

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#370]
While latest roughjs greatly improves the rendering, and virtually removes discrepancy between outline & BG (props to @pshihn), we still seem to have a problem around large ellipses, especially when such an ellipse is drawn in one step (i.e. not resized post-fact).

It happens even for smaller ellipses, but isn't as pronounced.

What this causes is that clicking on the edges of the ellipse doesn't select it, in case of underflow (is that a word?), or in case of overflow, it's selectable even outside visible boundary.

![image](https://user-images.githubusercontent.com/5153846/72335279-02b08e00-36bf-11ea-8c40-357713afcbd6.png)

![image](https://user-images.githubusercontent.com/5153846/72335340-1eb42f80-36bf-11ea-829e-6bc38230912a.png)

![image](https://user-images.githubusercontent.com/5153846/72335362-296ec480-36bf-11ea-9bfd-2cbeccafccce.png)

![image](https://user-images.githubusercontent.com/5153846/72335387-32f82c80-36bf-11ea-8218-2d80cd0a2d1d.png)

**Comment by pshihn** [comment]
I'd like to add that this is part of the sketchiness of roughjs. It offsets values randomly to create an imperfect look (e.g. fills getting outside the outline). So, to get the exact radii for an ellipse, though possible, is not what I intended to implement. 

Having said that, if this cannot be handled at the excalidraw app level, I'd be open to adding a custom render property when creating an ellipse, that tells the renderer to respect the ellipse radii values.

**Comment by dwelle** [comment]
Yea, I'm not sure what we want to achieve here, but I opened the issue to track this nonetheless. WTBS, isn't it possible to have exact radii, while retaining a sketch look? IMO the imperfect radius isn't what makes an element look sketchy.

**Comment by pshihn** [comment]
@dwelle  you are right. It has something to do with fill vs outline radii not being exact. 
I can think of multiple ways to do this in roughjs, and I do not want to make it the default option. I can add a parameter something like `persistDimensions: true` to achieve this.

**Comment by pshihn** [comment]
Ah I also noticed that I randomize the radii by up to 5% of its value, that's why it is more prominent in larger ellipses. Thanks for making me rethink this :)

**Comment by pshihn** [comment]
One of the reasons for making radii imperfect is that humans cannot sketch perfect circles, they always make it elliptical. I still think a bit of randomness is useful by default, but I will add a parameter to not randomize it.

**Comment by pshihn** [comment]
Added a config option to define how well curves are fitted: https://github.com/pshihn/rough/pull/126 This is available in v4.0.4 of roughjs
Wiki description: https://github.com/pshihn/rough/wiki#curvefitting

So when drawing an ellipse, your code can adapt to: 

```javascript
element.shape = generator.ellipse(
  element.width / 2,
  element.height / 2,
  element.width,
  element.height,
  {
     stroke: element.strokeColor,
     fill: element.backgroundColor === "transparent"
              ? undefined
              : element.backgroundColor,
     fillStyle: element.fillStyle,
     strokeWidth: element.strokeWidth,
     roughness: element.roughness,
     seed: element.seed,
     curveFitting: 1
   }
);
```

I'll leave it for someone to create a PR here :)

**Comment by dwelle** [comment]
Will do tomorrow. Thanks!
